crates/testing/src/task_helpers.rs

```diff
@@ -148,15 +235,67 @@ async fn build_quorum_proposal_and_signature(
         timestamp: 0,
         proposer_id: *api.public_key(),
     };
-    let signature = <BLSPubKey as SignatureKey>::sign(private_key, leaf.commit().as_ref())
+
+    let mut signature = <BLSPubKey as SignatureKey>::sign(private_key, leaf.commit().as_ref())
         .expect("Failed to sign leaf commitment!");
-    let proposal = QuorumProposal::<TestTypes> {
-        block_header,
-        view_number: ViewNumber::new(view),
+    let mut proposal = QuorumProposal::<TestTypes> {
+        block_header: block_header.clone(),
+        view_number: ViewNumber::new(1),
         justify_qc: QuorumCertificate::genesis(),
         timeout_certificate: None,
         proposer_id: leaf.proposer_id,
     };
+    for cur_view in 2..=view {
+        consensus.state_map.insert(
+            ViewNumber::new(1),
+            View {
+                view_inner: ViewInner::Leaf {
+                    leaf: leaf.commit(),
+                },
+            },
+        );
+        consensus.saved_leaves.insert(leaf.commit(), leaf.clone());
```

Comment:
Do we need to add to the saved leaves here?

Reply:
When creating the vote to trigger `QuorumVoteSend`, `build_vote()` is going to use `saved_leaves`.